pytups/superdict.py

```diff
@@ -454,16 +460,21 @@ def kapply(self, func: Callable, *args, **kwargs) -> "SuperDict":
         """
         return SuperDict({k: func(k, *args, **kwargs) for k in self})
 
-    def sapply(self, func: Callable, other: dict, *args, **kwargs) -> "SuperDict":
+    def sapply(
```

Comment:
for completion, should we also make it available for strings? After all, strings also allow for python operations such as `+` and `*`.

Reply:
I guess you mean when the other is also a string, right? and only allow it on `+` and `*`.

Now I am wondering what would happen when the value inside the dict is a string with some of these operations. I will add some tests to see if everything makes sense

Reply:
indeed, some combinations cannot be done (such as multiplying strings). Maybe not worth supporting it if it's going to fail without many checks. Up to you!